systems/plants/shapes/Geometry.h

```diff
@@ -0,0 +1,91 @@
+#ifndef __DrakeShapesGeometry_H__
+#define __DrakeShapesGeometry_H__
+
+#include <string>
+
+#include <Eigen/Dense>
+#include <Eigen/StdVector>
+
+#include "drakeShapesMacros.h"
+
+namespace DrakeShapes
+{
+  enum DLLEXPORT_drakeShapes Shape {
+    UNKNOWN,
+    BOX,
+    SPHERE,
+    CYLINDER,
+    MESH,
+    MESH_POINTS,
+    CAPSULE
+  };
+
+  class DLLEXPORT_drakeShapes Geometry {
+    public:
+      Geometry();
+      Geometry(const Geometry& other);
+
+      virtual Geometry* clone() const;
+
+      const Shape getShape() const;
+
+      // http://stackoverflow.com/questions/461203/when-to-use-virtual-destructors
+      virtual ~Geometry() {}
+
+    protected:
+      Geometry(Shape shape);
+      Shape shape;
+  };
+
+  class DLLEXPORT_drakeShapes Sphere: public Geometry {
+    public:
+      Sphere(double radius);
+      virtual Sphere* clone() const;
+      double radius;
+  };
+
+  class DLLEXPORT_drakeShapes Box : public Geometry {
+    public:
+      Box(const Eigen::Vector3d& size);
+      virtual Box* clone() const;
+      Eigen::Vector3d size;
+  };
+
+  class DLLEXPORT_drakeShapes Cylinder : public Geometry {
+    public:
+      Cylinder(double radius, double length);
+      virtual Cylinder* clone() const;
+      double radius;
+      double length;
+  };
+
+  class DLLEXPORT_drakeShapes Capsule : public Geometry {
+    public:
+      Capsule(double radius, double length);
+      virtual Capsule* clone() const;
+      double radius;
+      double length;
+  };
+
+  class DLLEXPORT_drakeShapes Mesh : public Geometry {
+    public:
+      Mesh(const std::string& filename);
+      Mesh(const std::string& filename, const std::string& resolved_filename);
+      virtual Mesh* clone() const;
+
+      std::string filename;
+      std::string resolved_filename;
```

Comment:
Is there a point to holding onto two file names here?  It looks like only one is ever used.